fiftyone/factory/repos/execution_store.py

```
@@ -570,6 +613,91 @@ def set_key(
     def set_cache_key(self, store_name, key, value, ttl=None):
         return self.set_key(store_name, key, value, ttl, policy="evict")
 
+    def set_key_if_absent(
+        self,
+        store_name,
+        key,
+        value,
+        ttl=None,
+        policy="persist",
+    ):
+        now = datetime.utcnow()
```

Ruff (0.16.2)

[warning] `datetime.datetime.utcnow()` used

(DTZ003)